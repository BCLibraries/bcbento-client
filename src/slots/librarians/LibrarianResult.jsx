import React from 'react';

function LibrarianResult({librarian}) {
    const url = `http://libguides.bc.edu/prf.php?account_id=${librarian.id}`;
    const img_url = `//${librarian.image}`;
    const alt = `picture of ${librarian.name}`;

    return <div className="media">
        <div className="media-left">
            <a href={url}><img src={img_url} alt={alt}/></a>
        </div>

        <div className="media-body">
            <h4 className="media-heading">
                <a href={url}>{librarian.name}</a>
            </h4>
            <p className="guide-description">{librarian.display_subjects}</p>
        </div>
    </div>;
}

export default LibrarianResult;