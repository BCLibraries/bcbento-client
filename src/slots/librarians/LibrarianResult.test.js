import renderer from 'react-test-renderer';
import LibrarianResult from './LibrarianResult.jsx';

it('renders correctly', () => {
    const librarian = {
        "id": "261",
        "name": "Steve Runge",
        "email": "steven.runge@bc.edu",
        "subjects": [
            "English Language and Literature"
        ],
        "image": "library.bc.edu/staff-portraits/steve-runge.jpg",
    };

    const component = renderer.create(
        <LibrarianResult librarian={librarian}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
