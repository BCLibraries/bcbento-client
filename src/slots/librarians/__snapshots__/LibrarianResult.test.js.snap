// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="media"
>
  <div
    className="media-left"
  >
    <a
      href="http://libguides.bc.edu/prf.php?account_id=261"
    >
      <img
        alt="picture of Steve Runge"
        src="//library.bc.edu/staff-portraits/steve-runge.jpg"
      />
    </a>
  </div>
  <div
    className="media-body"
  >
    <h3
      className="media-heading"
    >
      <a
        href="http://libguides.bc.edu/prf.php?account_id=261"
      >
        Steve Runge
      </a>
    </h3>
    <p
      className="guide-description"
    >
      English Language and Literature
    </p>
  </div>
</li>
`;
